### PR TITLE
Python bindings for torch_glow_training

### PR DESCRIPTION
**Pull request by putivsky**
Summary:
Refactored Python part of torch_glow into two parts sharing the same C++ code but independent in Python bindings. 
Documentation:

[Optional Fixes #issue]

Test Plan:

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

File: torch_glow/setup.py [hunk 1]
```diff
@@ -29,8 +29,12 @@
 print("torch version:", torch.__version__)
 print("torch location:", os.path.dirname(os.path.realpath(torch.__file__)))
 
+# Current setup.py file directory, i.e. glow/torch_glow.
 FILE_DIR = os.path.realpath(os.path.dirname(__file__))
-TOP_DIR = os.path.realpath(os.path.dirname(FILE_DIR))
+# Find the top directory with root Makefile, i.e. glow
+TOP_DIR = os.path.dirname(FILE_DIR)
```
Comment (jackm321) [inline torch_glow/setup.py]: whats this about?

File: torch_glow/src/binding.cpp [hunk 2]
```diff
@@ -23,6 +23,12 @@
 #include "PyTorchCommon.h"
 #include "PyTorchModelLoader.h"
 
+#include "TorchGlowTraining.h"
```
Comment (jackm321) [inline torch_glow/src/binding.cpp]: probably group this with the others above

File: torch_glow/src/binding.cpp [hunk 3]
```diff
@@ -23,6 +23,12 @@
...
+#include <pybind11/pybind11.h>
+/// Required include files for a proper binding TorchGlowTrainingWrapper class.
+#include <pybind11/stl.h>
+#include <torch/csrc/utils/pybind.h>
```
Comment (jackm321) [inline torch_glow/src/binding.cpp]: why is this needed?

File: torch_glow/src/CMakeLists.txt [hunk 4]
```diff
@@ -56,6 +57,7 @@ target_include_directories(PyTorchFileLoader PUBLIC
                             ${PYTORCH_DIR}/include)
 
 target_include_directories(_torch_glow PUBLIC
-                            ${PYTORCH_DIR}/include)
+                            ${PYTORCH_DIR}/include
+                            ${TORCH_GLOW}/src/training)
```
Comment (jackm321) [inline torch_glow/src/CMakeLists.txt]: is this also needed in addition to the target_link_libraries(...TorchGlowTraining...) ?

File: torch_glow/src/training/TorchGlowTraining.h [hunk 5]
```diff
@@ -93,6 +93,42 @@ class TorchGlowTraining {
...
+class TorchGlowTrainingWrapper {
+  // Trainer itself.
+  TorchGlowTraining trainer_;
+  // Required settings/parameters/configs.
+  TorchGlowTraining::ONNXWriterParameters parameters_;
+  PyTorchLoaderSettings settings_;
+  TrainingConfig config_;
+
+public:
+  TorchGlowTrainingWrapper() = default;
```
Comment (jackm321) [inline torch_glow/src/training/TorchGlowTraining.h]: not needed

File: torch_glow/src/training/TorchGlowTraining.h [hunk 6]
```diff
@@ -93,6 +93,42 @@ class TorchGlowTraining {
...
+  /// Converts PyTorch tensors into ATen inputs.
+  bool init(const std::string &modelPath,
```
Comment (jackm321) [inline torch_glow/src/training/TorchGlowTraining.h]: doxygen on the params and return value

File: torch_glow/src/training/TorchGlowTraining.h [hunk 7]
```diff
@@ -93,6 +93,42 @@ class TorchGlowTraining {
...
+
+  /// Saves Glow model into ONNX format.
+  bool save(const std::string &snapshotFile);
+
+  /// Helper functions for assigning settings/parameters/configs.
+  /// Sets PyTorchLoaderSettings
+  void setPyTorchLoaderSettings(bool enableFusionPass,
+                                bool enableWeightFreezing);
+
+  /// Sets ONNXWriterParameters
```
Comment (jackm321) [inline torch_glow/src/training/TorchGlowTraining.h]: periods

File: torch_glow/src/binding.cpp (hunk 3 above)
Comment (putivsky) [reply torch_glow/src/binding.cpp]: As comment says, without those two files, pybind doesn't recognize correspondently std::vector and at::Tensor types.

File: torch_glow/setup.py (hunk 1 above)
Comment (putivsky) [reply torch_glow/setup.py]: changing back.

File: torch_glow/src/CMakeLists.txt (hunk 4 above)
Comment (putivsky) [reply torch_glow/src/CMakeLists.txt]: yes, because glow doesn't follow fbcode standard for including paths (always from fbcode, even for local files), the additional include directories must be specified for the projects with more than one directory.
